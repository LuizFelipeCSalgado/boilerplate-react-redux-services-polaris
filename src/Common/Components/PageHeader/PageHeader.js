import React from 'react';
import PropTypes from 'prop-types';

import { Button } from '@shopify/polaris';

import './PageHeader.scss';

function PageHeader({ loadLogout }) {
  return (
    <div className="page-header">
      <div className="action-contant">
        <Button plain onClick={loadLogout}>Sair</Button>
      </div>
    </div>
  )
}

PageHeader.propTypes ={
  loadLogout: PropTypes.func.isRequired,
}

export default PageHeader;