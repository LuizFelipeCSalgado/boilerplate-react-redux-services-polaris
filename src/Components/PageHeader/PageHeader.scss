@import 'Common/Assets/Styles/variables.scss';

.page-header {
  background-color: $darkGray;
  height: 56px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  width: 100%;
  padding: 0px 10px;

  .Polaris-Button--plain {
    fill: $lightGray;
    color: $lightGray;
  }

  .Polaris-Button--plain:hover,
  .Polaris-Button--plain:focus,
  .Polaris-Button--plain:active {
    fill: $middleGray;
    color: $middleGray;
  }
}
