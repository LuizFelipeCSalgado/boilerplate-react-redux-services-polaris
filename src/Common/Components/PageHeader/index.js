import PageHeader from './PageHeader';
import { connect } from 'react-redux';

import { loadLogout } from 'Ducks/Auth';

const mapDispatchToProps = (dispatch) => ({
  loadLogout: () => { dispatch(loadLogout()) },
})

export default connect(
  null,
  mapDispatchToProps,
)(PageHeader);



