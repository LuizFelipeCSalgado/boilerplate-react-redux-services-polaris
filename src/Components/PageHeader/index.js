import { connect } from 'react-redux';

import { loadLogout } from 'Infrastructure/Ducks/Auth';
import PageHeader from './PageHeader';

const mapDispatchToProps = dispatch => ({
  loadLogout: () => {
    dispatch(loadLogout());
  },
});

export default connect(
  null,
  mapDispatchToProps,
)(PageHeader);
