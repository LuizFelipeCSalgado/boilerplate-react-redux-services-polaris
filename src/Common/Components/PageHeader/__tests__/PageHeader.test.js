import React from 'react';
import { mount } from 'enzyme';

import PageHeader from '../PageHeader';

describe('PageHeader', () => {
  let wrapper = null;
  const props = {};

  const mountComponent = (newProps = {}) => {
    wrapper = mount(<PageHeader {...props} {...newProps} />);
  };

  beforeEach(() => {
    mountComponent();
  });

  it('should render the without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

})
