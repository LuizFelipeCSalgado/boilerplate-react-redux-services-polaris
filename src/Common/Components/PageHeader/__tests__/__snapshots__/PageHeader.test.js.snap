// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should render the without crashing 1`] = `null`;
